spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/ClientManagerBackToBackTests.java

```diff
@@ -102,6 +131,40 @@ private void testSubscribeAndPublish(Class<?> configClass, String topicName, Cou
 		}
 	}
 
+	private void testSubscribeAndPublishRuntime(Class<?> configClass, String topicName, CountDownLatch subscribedLatch, Class<?> adapter)
+			throws Exception {
+
+		try (var ctx = new AnnotationConfigApplicationContext(configClass)) {
+			// given
+			var input = ctx.getBean("mqttOutFlow.input", MessageChannel.class);
+			var flowContext = ctx.getBean(IntegrationFlowContext.class);
+			var clientManager = ctx.getBean(ClientManager.class);
+			var output = new QueueChannel();
+			Class<?>[] parameterTypes = {ClientManager.class, String[].class};
+			Constructor<?> declaredConstructor = adapter.getConstructor(parameterTypes);
```

Comment:
I don't see a reason in this reflection.
You just can have a factory method in the `Mqttv5ConfigRuntime` which would return us a respective `MessageProducerSupport` instance.
The `@Configuration` class is registered as a bean as well.
So, you might just need to extract some common interface for both of your `Mqttv5ConfigRuntime` and `Mqttv3ConfigRuntime` to have that factory method implemented respectively.